src/util/misc.rs

```diff
@@ -63,6 +67,8 @@ mod message_signing {
                 MessageSignatureError::InvalidLength => write!(f, "length not 65 bytes"),
                 MessageSignatureError::InvalidEncoding(ref e) => write!(f, "invalid encoding: {}", e),
                 MessageSignatureError::InvalidBase64 => write!(f, "invalid base64"),
+                MessageSignatureError::UnableToRecoverPubKey(ref e) => write!(f, "error recovering public key: {}", e),
+                MessageSignatureError::UnsupportedAddressType(ref e) => write!(f, "unsupported address type: {}", e),
```

Comment:
The inner type is not an error but an `AddressType`, perhaps `ty` or `addr` would be better than `e`?

Reply:
Wouldn't an explicit map be better here? It helps differentiate between `InvalidEncoding(secp256k1::Error)` and `UnableToRecoverPubKey(secp256k1::Error)`

Reply:
I believe @tcharding is only pointing out variable naming. Not sure what do you mean by map.

Reply:
Yes good catch. I hope an explicit `address_type` is acceptable.

Reply:
@Kixunil I was distracted and posted my reply to a different comment here. You can ignore. I immediately deleted the comment, but you are too fast.